### PR TITLE
HPCC-29541 Fix ZAP report postmortem directory structure

### PR DESCRIPTION
**Pull request by jakesmith**
Ensure the postmortem directory structure is preserved in the ZAP report.
This prevents multiple component postmortem repots being folded and potentially overwriting each other in the report.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by wangkx — COMMENTED** [review]
@jakesmith should some changes be made in the ZAPReport Import function to match with the new directory structure in a ZAPReport? When importing a WU from a ZAP report, the existing ZAPReport Import function verifies the postmortem folder using ImportDIR + the Tail of the @filename in WU XML (see the updateWUQueryAssociatedFilesAttrs() in workunit.cpp). Now, the verification will not work because a subFolder is added between the ImportDIR and  the Tail.

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -5412,8 +5413,7 @@ void CWorkUnitFactory::importWorkUnit(const char *zapReportFileName, const char
                 throw MakeStringException(WUERR_InvalidUserInput, "Failed to find %s in %s.", mask, unzipDir.str());
             }
 
-            fileName.set(unzipDir).append(PATHSEPSTR);
-            di->getName(fileName);
+            fileName.set(di->query().queryFilename());
```
Comment (jakesmith) [inline common/workunit/workunit.cpp:5416]: NB: this is an incidental change, but cleaner and relates to removal of a spurious extra PATHSEPCHAR in the unzipDir (removed in another change).

File: common/workunit/workunit.cpp [hunk 2]
```diff
@@ -5474,7 +5491,8 @@ void CWorkUnitFactory::importWorkUnit(const char *zapReportFileName, const char
         }
         void setUNZIPDir()
         {   //Set a unique unzip folder inside the component's data folder
-            unzipDir.append(importDir).append(PATHSEPSTR).append(wuid.get());
+            unzipDir.append(importDir);
+            addPathSepChar(unzipDir).append(wuid.get());
```
Comment (jakesmith) [inline common/workunit/workunit.cpp:5495]: NB: somewhat unrelated, but prior to this change, there would be a double // created in the path.

**Comment by jakesmith** [comment]
> @jakesmith should some changes be made in the ZAPReport Import function to match with the new directory structure in a ZAPReport? When importing a WU from a ZAP report, the existing ZAPReport Import function verifies the postmortem folder using ImportDIR + the Tail of the @filename in WU XML (see the updateWUQueryAssociatedFilesAttrs() in workunit.cpp). Now, the verification will not work because a subFolder is added between the ImportDIR and the Tail.

Yes, it needed reworking. Please see new commit and explanation within the commit comment.

**Review by wangkx — APPROVED** [review]
@jakesmith looks fine now

**Review by ghalliday — CHANGES_REQUESTED** [review]
@jakesmith the windows build failure looks like a real error.

**Comment by jakesmith** [comment]
@ghalliday - windows build issue resolved.

**Review by ghalliday — APPROVED** [review]
@jakesmith looks reasonable, no problems spotted.  Please squash.

**Comment by jakesmith** [comment]
> @jakesmith looks reasonable, no problems spotted. Please squash.

@ghalliday - now squashed.

**Pull request merged**